tools/setup/emoji/generate_emoji_names

```diff
@@ -72,9 +108,47 @@ def main() -> None:
             all_canonical_names.add(canonical_name)
             all_emojis[emoji_code] = CUSTOM_EMOJI_NAME_MAPS[emoji_code]
         else:
-            continue  # this commit doesn't add CLDR data yet.
             # create the unicode character(s) for the emoji, since this is the key into the CLDR data
+            emoji = "".join(
+                chr(int(h, 16))
+                for h in (emoji_dict["non_qualified"] or emoji_dict["unified"]).split("-")
+            )
+            if emoji not in CLDR_DATA:
+                print(
+                    f"{emoji} not found in custom emoji name maps, but also not found in CLDR data. Skipping."
+                )
+                continue
+            assert len(CLDR_DATA[emoji]["tts"]) == 1
+            canonical_name = cleanup_name(CLDR_DATA[emoji]["tts"][0].strip())
+            if canonical_name in all_canonical_names:
+                raise Exception(
+                    f"{canonical_name} was already added with a different codepoint. "
+                    f"Rename it in `custom_emoji_names` or add an entry for {emoji_code}."
+                )
+            aliases = [cleanup_name(alias.strip()) for alias in CLDR_DATA[emoji]["default"]]
+            all_emojis[emoji_code] = {"canonical_name": canonical_name, "aliases": aliases}
+            all_canonical_names.add(canonical_name)
+
+    # STEP 2: We don't support having aliases that collide with canonical names for emoji, so remove them.
+    for (emoji_code, emoji_names) in all_emojis.items():
+        aliases = emoji_names["aliases"][
+            :
+        ]  # Copy the list to not iterate while elements are being deleted.
+        for alias in aliases:
+            if alias in all_canonical_names:
+                emoji_names["aliases"].remove(alias)
+            else:
+                alias_to_emoji_code[alias].append(emoji_code)  # Preparation for STEP 3.
+
+    # STEP 3: We don't support having aliases for multiple emoji. In these situations, remove the alias
+    # for all relevant emoji (unless they show up in custom_emoji_names).
+    for alias in alias_to_emoji_code.keys():
+        if len(alias_to_emoji_code[alias]) > 1:
+            for emoji_code in alias_to_emoji_code[alias]:
+                if emoji_code not in CUSTOM_EMOJI_NAME_MAPS:
+                    all_emojis[emoji_code]["aliases"].remove(alias)
```

Comment:
Is the intent that we'd resolve this sort of conflict via `CUSTOM_EMOJI_NAME_MAPS` in general? If they're uncommon, I'm kinda wondering if it'd make sense to have an assert statement here instead.

Reply:
I can add a comment here for why we needed to add this, but essentially the CLDR data doesn't have the same sense of unique alias that we do, so we have to filter it out ourselves. I don't want to hardcode it all in because we should refer to CLDR data as a point of truth as much as possible. If the alias was specifically specified in the custom list then we can keep just that one, but otherwise I just removed all of them instead of randomly picking a favourite that gets to use it. Does that make sense?

Reply:
The comment here is very helpful.